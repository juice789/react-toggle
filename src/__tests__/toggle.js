/**
 * @jest-environment jsdom
 */

import React from 'react'
import { render } from '@testing-library/react'
import 'jest-styled-components'

import Toggle from '../index'

it('default component', () => {
  const { container } = render(<Toggle />)
  expect(container.firstChild).toMatchSnapshot()
})

it('set to on', () => {
  const { container } = render(<Toggle isOn={true} />)
  expect(container.firstChild).toMatchSnapshot()
})

it('controlled', () => {
  let val = true
  const {
    container: { firstChild },
    rerender
  } = render(<Toggle value={val} />)
  expect(firstChild).toMatchSnapshot()
  const Thumb = firstChild.querySelector('div').querySelector('div').firstChild
  expect(Thumb).toHaveStyleRule('transform', 'translateX( calc(100% + 0px) )')
  val = false
  rerender(<Toggle value={val} />)
  expect(Thumb).toHaveStyleRule('transform', 'translateX( 0% )')
})

it('custom styles', () => {
  const styles = {
    track: (defaults) => ({
      ...defaults,
      background: 'green'
    }),
    thumb: (defaults) => ({
      ...defaults,
      fontSize: '20px'
    })
  }
  const {
    container: { firstChild }
  } = render(<Toggle text={'YO'} styles={styles} />)

  expect(firstChild).toMatchSnapshot()
})

it('custom components', () => {
  const {
    container: { firstChild }
  } = render(
    <Toggle text={'HI'}>
      <Toggle.Left>1</Toggle.Left>
      <Toggle.Right>2</Toggle.Right>
    </Toggle>
  )
  expect(firstChild).toMatchSnapshot()
})
