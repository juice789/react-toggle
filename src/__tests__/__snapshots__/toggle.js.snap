// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controlled 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #3a3747;
  border-radius: 25px;
  width: calc(25px * 2);
  height: 25px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border: 1px solid #3a3747;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
}

.c1:hover {
  background: #3a3747;
  border: 1px solid #6e66a6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25px;
  height: 25px;
}

.c3 {
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateX( calc(100% + 0px) );
  -ms-transform: translateX( calc(100% + 0px) );
  transform: translateX( calc(100% + 0px) );
  text-align: center;
  background: #6e66a6;
  width: calc(25px - 0px);
  height: calc(25px - 0px);
  color: #faf9f8;
  border-radius: 25px;
  font-size: 0.7rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 50%;
  height: 100%;
  font-size: 0.7rem;
  color: #efedeb;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        on="1"
      />
    </div>
    <div
      class="c4"
    />
    <div
      class="c4"
    />
  </div>
</div>
`;

exports[`custom components 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #3a3747;
  border-radius: 25px;
  width: calc(25px * 2);
  height: 25px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border: 1px solid #3a3747;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
}

.c1:hover {
  background: #3a3747;
  border: 1px solid #6e66a6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25px;
  height: 25px;
}

.c3 {
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateX( 0% );
  -ms-transform: translateX( 0% );
  transform: translateX( 0% );
  text-align: center;
  background: #6e66a6;
  width: calc(25px - 0px);
  height: calc(25px - 0px);
  color: #faf9f8;
  border-radius: 25px;
  font-size: 0.7rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 50%;
  height: 100%;
  font-size: 0.7rem;
  color: #efedeb;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        on="0"
      >
        HI
      </div>
    </div>
    <div
      class="c4"
    >
      1
    </div>
    <div
      class="c4"
    >
      2
    </div>
  </div>
</div>
`;

exports[`custom styles 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: green;
  border-radius: 25px;
  width: calc(25px * 2);
  height: 25px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border: 1px solid #3a3747;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
}

.c1:hover {
  background: #3a3747;
  border: 1px solid #6e66a6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25px;
  height: 25px;
}

.c3 {
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateX( 0% );
  -ms-transform: translateX( 0% );
  transform: translateX( 0% );
  text-align: center;
  background: #6e66a6;
  width: calc(25px - 0px);
  height: calc(25px - 0px);
  color: #faf9f8;
  border-radius: 25px;
  font-size: 20px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 50%;
  height: 100%;
  font-size: 0.7rem;
  color: #efedeb;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        on="0"
      >
        YO
      </div>
    </div>
    <div
      class="c4"
    />
    <div
      class="c4"
    />
  </div>
</div>
`;

exports[`default component 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #3a3747;
  border-radius: 25px;
  width: calc(25px * 2);
  height: 25px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border: 1px solid #3a3747;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
}

.c1:hover {
  background: #3a3747;
  border: 1px solid #6e66a6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25px;
  height: 25px;
}

.c3 {
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateX( 0% );
  -ms-transform: translateX( 0% );
  transform: translateX( 0% );
  text-align: center;
  background: #6e66a6;
  width: calc(25px - 0px);
  height: calc(25px - 0px);
  color: #faf9f8;
  border-radius: 25px;
  font-size: 0.7rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 50%;
  height: 100%;
  font-size: 0.7rem;
  color: #efedeb;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        on="0"
      />
    </div>
    <div
      class="c4"
    />
    <div
      class="c4"
    />
  </div>
</div>
`;

exports[`set to on 1`] = `
.c0 {
  position: relative;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #3a3747;
  border-radius: 25px;
  width: calc(25px * 2);
  height: 25px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  cursor: pointer;
  border: 1px solid #3a3747;
  -webkit-transition: background 0.2s ease;
  transition: background 0.2s ease;
}

.c1:hover {
  background: #3a3747;
  border: 1px solid #6e66a6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 25px;
  height: 25px;
}

.c3 {
  -webkit-transition: -webkit-transform 0.2s ease;
  -webkit-transition: transform 0.2s ease;
  transition: transform 0.2s ease;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transform: translateX( calc(100% + 0px) );
  -ms-transform: translateX( calc(100% + 0px) );
  transform: translateX( calc(100% + 0px) );
  text-align: center;
  background: #6e66a6;
  width: calc(25px - 0px);
  height: calc(25px - 0px);
  color: #faf9f8;
  border-radius: 25px;
  font-size: 0.7rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 50%;
  height: 100%;
  font-size: 0.7rem;
  color: #efedeb;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
        on="1"
      />
    </div>
    <div
      class="c4"
    />
    <div
      class="c4"
    />
  </div>
</div>
`;
